src/mscorlib/src/System/Guid.cs

```diff
@@ -45,18 +45,22 @@ public struct Guid : IFormattable, IComparable
         ////////////////////////////////////////////////////////////////////////////////
 
         // Creates a new guid from an array of bytes.
-        //
         public Guid(byte[] b)
+            : this(new ReadOnlySpan<byte>(b))
+        {
+        }
+
+        // Creates a new guid from a read-only span.
+        public Guid(ReadOnlySpan<byte> b)
         {
             if (b == null)
                 throw new ArgumentNullException(nameof(b));
```

Comment:
This check for b == null needs to be done in the Guid(byte[]) ctor, not in the span-based ctor.  I'd suggest removing this if/throw from here, and changing the code above from:
```C#
this(new ReadOnlySpan<byte>(b))
```
to instead be:
```C#
this(new ReadOnlySpan<byte>(b ?? throw new ArgumentNullException(nameof(b)))
```